src/emqx_protocol.erl

```diff
@@ -204,26 +204,27 @@ received(?PACKET(?CONNECT), PState = #pstate{connected = true}) ->
 
 received(Packet = ?PACKET(Type), PState) ->
     trace(recv, Packet, PState),
-    case catch emqx_packet:validate(Packet) of
+    try emqx_packet:validate(Packet) of
         true ->
             {Packet1, PState1} = preprocess_properties(Packet, PState),
-            process_packet(Packet1, inc_stats(recv, Type, PState1));
-        {'EXIT', {protocol_error, _Stacktrace}, PState} ->
+            process_packet(Packet1, inc_stats(recv, Type, PState1))
+    catch
+        error:protocol_error:_Stacktrace ->
```

Comment:
if it is to discard stacktrace like this (`_Stacktrace`).
there is no need to bind it.

Reply:
Yes